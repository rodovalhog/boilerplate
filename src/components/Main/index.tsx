import * as S from './styles'

const Main = ({
  title = 'React AvanÃ§ado',
  description = 'TypeScript, React NextJS, Styled-Components'
}) => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo-react.png"
      alt="Imagem de um atomo simbolo do react"
      height="100px"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="/img/Minato_Base.png"
      alt="Imagem Minato namikase personagem ficticio do anime japones naruto "
      height="400px"
    />
  </S.Wrapper>
)
export default Main
