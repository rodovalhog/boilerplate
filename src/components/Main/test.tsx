import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the hediing', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot() // esse cara é pra gerar o snapshot
  })

  it('should render the color correclty', () => {
    const { container } = render(<Main />)
    //verifica o estilo que está no componente
    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b'
    })
  })
})
