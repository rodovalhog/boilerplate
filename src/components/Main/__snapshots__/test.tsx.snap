// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the hediing 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
}

<main
  class="c0"
>
  <img
    alt="Imagem de um atomo simbolo do react"
    class="c1"
    height="100px"
    src="/img/logo-react.png"
  />
  <h1
    class="c2"
  >
    React Avan√ßado
  </h1>
  <h2
    class="c3"
  >
    TypeScript, React NextJS, Styled-Components
  </h2>
  <img
    alt="Imagem Minato namikase personagem ficticio do anime japones naruto "
    class="c4"
    height="400px"
    src="/img/Minato_Base.png"
  />
</main>
`
